### PR TITLE
Create functions for getting large scenery type and sequence

### PR DESCRIPTION
**Pull request by Gymnasiast**
Taken from #6523.

### CODE REVIEW

File: src/openrct2/paint/tile_element/scenery_multiple.c [hunk 1]
```diff
@@ -186,8 +186,8 @@ static const boundbox s98E3C4[] = {
 void scenery_multiple_paint(paint_session * session, uint8 direction, uint16 height, rct_tile_element *tileElement) {
     //RCT2_CALLPROC_X(0x6B7F0C, 0, 0, direction, height, (sint32)tileElement, 0, 0); return;
     session->InteractionType = VIEWPORT_INTERACTION_ITEM_LARGE_SCENERY;
-    uint32 ebp = tileElement->properties.scenerymultiple.type >> 10;
-    rct_scenery_entry *entry = get_large_scenery_entry(tileElement->properties.scenerymultiple.type & 0x3FF);
+    uint32 ebp = scenery_large_get_sequence(tileElement);
```
Comment (duncanspumpkin) [inline src/openrct2/paint/tile_element/scenery_multiple.c]: Rename ebp to sequence num. Remove above callproc comment

File: src/openrct2/paint/tile_element/scenery_multiple.c [hunk 2]
```diff
@@ -237,7 +237,7 @@ void scenery_multiple_paint(paint_session * session, uint8 direction, uint16 hei
     }
     if (entry->large_scenery.flags & LARGE_SCENERY_FLAG_3D_TEXT) {
         if (entry->large_scenery.tiles[1].x_offset != (sint16)(uint16)0xFFFF) {
-            sint32 al = ((tileElement->properties.surface.terrain >> 2) - 1) & 3;
+            sint32 al = (scenery_large_get_sequence(tileElement) - 1) & 3;
```
Comment (duncanspumpkin) [inline src/openrct2/paint/tile_element/scenery_multiple.c]: Label al sequence num

File: src/openrct2/paint/tile_element/scenery_multiple.c [hunk 3]
```diff
@@ -329,7 +329,7 @@ void scenery_multiple_paint(paint_session * session, uint8 direction, uint16 hei
         scenery_multiple_paint_supports(session, direction, height, tileElement, dword_F4387C, tile);
         return;
     }
-    uint8 al = ((tileElement->properties.surface.terrain >> 2) - 1) & 3;
+    uint8 al = (scenery_large_get_sequence(tileElement) - 1) & 3;
```
Comment (duncanspumpkin) [inline src/openrct2/paint/tile_element/scenery_multiple.c]: As above

File: src/openrct2/windows/EditorObjectSelection.cpp [hunk 4]
```diff
@@ -559,7 +559,7 @@ static void setup_in_use_selection_flags()
             Editor::SelectedObjects[OBJECT_TYPE_WALLS][type] |= (1 << 0);
             break;
         case TILE_ELEMENT_TYPE_SCENERY_MULTIPLE:
-            type = iter.element->properties.scenerymultiple.type & 0x3FF;
+            type = scenery_large_get_sequence(iter.element);
```
Comment (duncanspumpkin) [inline src/openrct2/windows/EditorObjectSelection.cpp]: Wrong function

File: src/openrct2/windows/TileInspector.cpp [hunk 5]
```diff
@@ -1874,15 +1874,15 @@ static void window_tile_inspector_paint(rct_window *w, rct_drawpixelinfo *dpi)
         {
             // Details
             // Type
-            sint16 largeSceneryType = tileElement->properties.scenerymultiple.type & 0x03FF;
+            sint16 largeSceneryType = scenery_large_get_type(tileElement);
             gfx_draw_string_left(dpi, STR_TILE_INSPECTOR_LARGE_SCENERY_TYPE, &largeSceneryType, COLOUR_DARK_GREEN, x, y);
 
             // Part ID
-            sint16 pieceID = (tileElement->properties.scenerymultiple.type & 0xFC00) >> 10;
+            sint16 pieceID = scenery_large_get_sequence(tileElement);
```
Comment (duncanspumpkin) [inline src/openrct2/windows/TileInspector.cpp]: We should rename everything to piece id our sequence and adjust the code to suit

File: src/openrct2/world/map.c [hunk 6]
```diff
@@ -2728,10 +2728,13 @@ void game_command_place_large_scenery(sint32* eax, sint32* ebx, sint32* ecx, sin
         rct_tile_element* tile_element = map_get_surface_element_at(curTile.x / 32, curTile.y / 32);
         if(tile_element != NULL){
             sint32 height = tile_element->base_height * 8;
+            sint32 type = scenery_large_get_type(tile_element);
 
-            if(tile_element->properties.scenerymultiple.type & 0xF){
+            if (type & 0xF)
```
Comment (duncanspumpkin) [inline src/openrct2/world/map.c]: How odd this code is. Why would a type also encode height info. I want to double check the assembly for this section. Need to update wiki as well about this info.

File: src/openrct2/windows/TileInspector.cpp (hunk 5 above)
Comment (Gymnasiast) [reply src/openrct2/windows/TileInspector.cpp]: What do you mean?

File: src/openrct2/world/map.c (hunk 6 above)
Comment (Gymnasiast) [reply src/openrct2/world/map.c]: Oh, I think I know!
It gets `tile_element` by doing `map_get_surface_element_at()`. So this isn't a large scenery object at all.It's a surface element, and the property at this location is `slope`.

Comment (Gymnasiast) [reply src/openrct2/world/map.c]: https://github.com/OpenRCT2/OpenRCT2/pull/6684/files

**Pull request merged**